movement/movement.py

```diff
@@ -0,0 +1,366 @@
+    
+# Combined motor functions
+def drive_motor(left : GPIO.PWM, right: GPIO.PWM, speed, direction):
+    if speed > 0:
+        driving_function[direction](left, right, speed)
+    else:
+        driving_function[direction-1](left, right, speed)
+
+def drive_all(direction, front_left_spd, front_right_spd, back_left_spd, back_right_spd):
+    drive_motor(FLL, FLR, front_left_spd, direction)
+    drive_motor(FRL, FRR, front_right_spd, direction)
+    # drive_motor(BLL, BLR, back_left_spd, direction)
+    # drive_motor(BRL, BRR, back_right_spd, direction)
+
+# side = 0 # 0 is right, 1 is left # Single side ultrasonic
+
+sensors = (side_front, side_back)
+
+TARGET_DIST_IN = 3
+TARGET_DIST_CM = TARGET_DIST_IN * 2.54
+SENSOR_GAP_IN = 13 # IN
+SENSOR_GAP_CM = SENSOR_GAP_IN * 2.54
+BASE_SPEED = 25
+motor_speeds = [BASE_SPEED, BASE_SPEED, BASE_SPEED, BASE_SPEED] # FL, FR, BL, BR
+
+# speeds = [BASE_SPEED, BASE_SPEED] # DIFFERENTIAL: left, right
+
+cycle = 0
+
+# previous_dist = sensors[side].distance # Single side Ultrasonic
+
+#####################
+# Encoder Setup #
+#####################
+
+# Pins
+encoder_pin_a = 2
+encoder_pin_b = 3
+
+encoder_value = 0
+encoder_last_state = 0
+
+GPIO.setup(encoder_pin_a, GPIO.IN, pull_up_down=GPIO.PUD_UP)
+GPIO.setup(encoder_pin_b, GPIO.IN, pull_up_down=GPIO.PUD_UP)
+
+total_distance = 0
+WHEEL_DIA = 2.7559
+WHEEL_CIRCUMFERENCE = WHEEL_DIA * math.pi
+
+# updating encoder and distance
+def update_encoder():
+    global encoder_value, encoder_last_state, total_distance
+
+    # Read the current state of the encoder pins
+    current_state_a = GPIO.input(encoder_pin_a)
+    current_state_b = GPIO.input(encoder_pin_b)
+
+    side_front_dist = sensors[direction].distance
+    side_back_dist = sensors[direction-1].distance
+
+    # Combine the current states into a single value
+    current_state = (current_state_a << 1) | current_state_b
+
+    # Determine the direction of rotation based on the previous and current states
+    direction = (current_state - encoder_last_state) % 4
+
+    current_rad_angle = math.acos((side_back_dist - side_front_dist) / SENSOR_GAP_CM)
+
+    # Update the encoder value based on the direction
+    if direction == 1 or direction == -3:
+        encoder_value += 1
+        total_distance += WHEEL_CIRCUMFERENCE / 4 * math.sin(current_rad_angle)
+    elif direction == -1 or direction == 3:
+        encoder_value -= 1
+        total_distance -= WHEEL_CIRCUMFERENCE / 4 * math.sin(current_rad_angle)
+
+    # Update the last state for the next iteration
+    encoder_last_state = current_state
+
+    print(f"Total distance from start: {total_distance // 12} ft {total_distance % 12} in")
+
+# Update encoder value on change
+GPIO.add_event_detect(encoder_pin_a, GPIO.BOTH, callback=update_encoder)
+GPIO.add_event_detect(encoder_pin_b, GPIO.BOTH, callback=update_encoder)
+
+########
+# Main #
+########
+
+def auto_drive():
+    try:
+        while True:
+            # Swerve drive
+            
```

Comment:
Take out the while loop because we will have one in main for auto drive. It will be dependent on a variable mutation via the website.